### PR TITLE
main: fix join failed when log dir is same as data dir

### PR DESCRIPTION
**Pull request by Connor1996**
if log directory is subdirectory of data directory, a new pd to join cluster will fail cause log directory will create before checking data directory whether exists and pd will regard the new join pd as a already existing pd.

this pr does: 
don't allow log directory to be the subdirectory of data directory

### CODE REVIEW

File: server/join.go [hunk 1]
```diff
@@ -72,9 +73,8 @@ func PrepareJoinCluster(cfg *Config) error {
 	}
 
 	// Cases with data directory.
-
 	initialCluster := ""
-	if wal.Exist(cfg.DataDir) {
+	if wal.Exist(path.Join(cfg.DataDir, "member")) {
```
Comment (disksing) [inline server/join.go]: How about check if the directory is empty instead?

Comment (Connor1996) [reply server/join.go]: if empty, wal.Exist() will return false.

File: cmd/pd-server/main.go [hunk 2]
```diff
@@ -50,6 +52,13 @@ func main() {
 		log.Fatalf("parse cmd flags error: %s\n", err)
 	}
 
+	dataDir, err := filepath.Abs(cfg.DataDir)
+	logFile, err := filepath.Abs(cfg.Log.File.Filename)
+	rel, err := filepath.Rel(dataDir, filepath.Dir(logFile))
+	if !strings.HasPrefix(rel, "..") {
```
Comment (siddontang) [inline cmd/pd-server/main.go]: sometimes user may not use `..` here.

Comment (Connor1996) [reply cmd/pd-server/main.go]: Rel() gets relative path based on absolute path, it has nothing to do with whether user use ".." or not

**Pull request merged**